### PR TITLE
new:dev: Upgrade thrift & maven-thrift-plugin. Make tests relocatable

### PR DESCRIPTION
**Pull request by vrajat**
Upgraded thrift from 0.9.0 to 0.9.3. The main motivation is that
0.9.3 can be installed through brew on mac. I tried to upgrade to
thrift 0.10.0 but there were test failures.

Upgrade maven-thrift-plugin to 0.1.11. 0.1.10 is not available anymore
in http://maven.davidtrott.com/repository. 0.1.11 is available in apache
maven repository.
   
Another improvement is that tests assumed specific directory
structures internal to Qubole. The test directories are now
configured relative to `java.io.tmpdir`.

### CODE REVIEW

File: rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/LocalDataTransferServer.java [hunk 1]
```diff
@@ -106,7 +106,8 @@ public void run()
                 log.info("Stopping Local Transfer server");
             }
             catch (IOException e) {
-                log.error(String.format("Error starting Local Transfer server %s", e));
+                log.error(String.format("Error starting Local Transfer server %s", e), e);
+                e.printStackTrace();
```
Comment (stagraqubole) [inline rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/LocalDataTransferServer.java]: wasnt log logging exceptions? We are using the format where exception should be logged

File: rubix-tests/src/test/java/com/qubole/rubix/tests/TestNonLocalReadRequestChain.java [hunk 2]
```diff
@@ -32,28 +32,45 @@
 import org.apache.hadoop.fs.Path;
 import org.apache.log4j.Level;
 import org.apache.log4j.Logger;
-import org.testng.annotations.AfterMethod;
-import org.testng.annotations.BeforeMethod;
-import org.testng.annotations.Test;
+import org.testng.annotations.*;
```
Comment (stagraqubole) [inline rubix-tests/src/test/java/com/qubole/rubix/tests/TestNonLocalReadRequestChain.java]: Didnt checkstyle crib about this compaction? 
Easier to read import without compacting them to '*', we should revert this.

**Comment by stagraqubole** [comment]
lgtm, couple of minor comments.

File: rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/LocalDataTransferServer.java (hunk 1 above)
Comment (vrajat) [inline rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/LocalDataTransferServer.java]: It does but just the message. I wanted stack trace but this didnt work out as well. I'll revert this change.

File: rubix-tests/src/test/java/com/qubole/rubix/tests/TestNonLocalReadRequestChain.java (hunk 2 above)
Comment (vrajat) [inline rubix-tests/src/test/java/com/qubole/rubix/tests/TestNonLocalReadRequestChain.java]: No. it didnt. Test files are ignored ?

Comment (stagraqubole) [inline rubix-tests/src/test/java/com/qubole/rubix/tests/TestNonLocalReadRequestChain.java]: Ya, looks like it. Tried and it fails for main code and passes for tests.

**Pull request merged**